commerce_coordinator/apps/titan/clients.py

```diff
@@ -0,0 +1,104 @@
+"""
+API clients for Titan.
+"""
+from urllib.parse import urljoin
+
+import requests
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from edx_rest_api_client.client import OAuthAPIClient
+
+# Use special Celery logger for tasks client calls.
+logger = get_task_logger(__name__)
+
+
+class TitanAPIClient():
+    """
+    API client for calls to Titan using API key.
+    """
+
+    def __init__(self):
+        self.client = requests.Session()
+        # Always send API key.
+        self.client.headers.update(self.api_key_header)
+
+    @property
+    def api_base_url(self):
+        """URL of API service."""
+        return urljoin(settings.TITAN_URL, '/v1')
+
+    @property
+    def api_key_header(self):
+        """Header to add to all API requests to authenticate to endpoint."""
+        return {'X-API-Key': settings.TITAN_API_KEY}
```

Comment:
Do you specify the `Content-Type` of the header somewhere?

Reply:
No, but it's autodetected I think. Here's a sample response, see the `Content-Type` at the end.

```
[2022-09-06 12:58:26,612: DEBUG/ForkPoolWorker-8] commerce_coordinator.apps.titan.tasks.enrollment_code_redemption_requested_create_order_task[459b0255-a169-413e-a419-a012ce11fd82]: Request headers: {'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'X-API-Key': 'replace-me', 'Content-Length': '114', 'Content-Type': 'application/json'}
```

Reply:
`Header to add to all API requests` - it would be to all API requests to Titan only right?

Reply:
Done: https://github.com/edx/commerce-coordinator/pull/46/commits/6c4cf3f3f6682371c8554f3533ac4dcabcf5f2a5